mitxgraders/baseclasses.py

```diff
@@ -205,6 +207,60 @@ def log_output(self):
         content = "\n".join(self.debuglog)
         return "<pre>{content}</pre>".format(content=content)
 
+    @classmethod
+    def ensure_text_inputs(cls, student_input, allow_lists=True, allow_single=True):
+        """
+        Ensures that student_input is a list of text strings.
+
+        NOTE: Overriden by ItemGrader to ensure input is a single text string.
+        """
+        list_errmsg = None
+        single_errmsg = None
+
+        if allow_lists and isinstance(student_input, list):
+            try:
+                return Schema([text_string])(student_input)
+            except MultipleInvalid as error:
+                pos = error.path[0]
+                list_errmsg = (
+                    "Expected a list of text strings for student_input, but "
+                    "item at position {pos} has {thetype}"
+                    .format(pos=pos, thetype=type(student_input[pos])))
+
+        if allow_lists and not isinstance(student_input, list):
```

Comment:
Just cleaning this routine up a bit. I'd move this if statement below the `if allow_single:` below. See next comment.